File: tests/06_phases_ompss.dg/cxx/success_task_04.cpp
/*
<testinfo>
test_generator=config/mercurium-ompss
</testinfo>
*/

void f1(int &a, int &b, int &c, int &d)
{
#pragma omp task inout(a) shared(b) private(c) firstprivate(d)
    {
        c = 3;
        a++;
        b++;
        d++;
    }
#pragma omp taskwait
}

int main(int argc, char* argv[])
{
    int x, y, z, k;

    f1(x, y, z, k);

    return 0;
}

File: tests/07_phases_omp.dg/common/fortran/success_task_04.f90
! <testinfo>
! test_generator=config/mercurium-omp
! </testinfo>
MODULE M1
  IMPLICIT NONE
  TYPE T1
    INTEGER :: X
  END TYPE T1
END MODULE M1

MODULE M2
  USE M1
  IMPLICIT NONE

   TYPE T2
       TYPE(T1) :: T1
   END TYPE T2

  TYPE(T2) :: VAR
END MODULE M2

PROGRAM MAIN
   USE M2
   IMPLICIT NONE

   VAR % T1 % X = 0

   !$OMP TASK SHARED(VAR)
       VAR % T1 % X  = 42
   !$OMP END TASK
   !$OMP TASKWAIT

   IF (VAR % T1 % X /= 42) STOP 1
END PROGRAM MAIN
